src/react/hooks.js

```diff
@@ -48,3 +52,298 @@ export const useTrackColorSchemeChoice = () => {
     };
   }, []);
 };
+
+/**
+ * Adds/updates a `<link>` element in the HTML document to load the core application theme CSS.
+ *
+ * @memberof module:React
+ * @param {object} args
+ * @param {object} args.themeCore Object representing the core Paragon theme CSS.
+ * @param {string} args.onLoad A callback function called when the core theme CSS is loaded.
+ */
+export const useParagonThemeCore = ({
+  themeCore,
+  onLoad,
+}) => {
+  useEffect(() => {
+    // If there is no config for the core theme url, do nothing.
+    if (!themeCore?.url) {
+      return;
+    }
+    let coreThemeLink = document.head.querySelector(`link[href='${themeCore.url}']`);
+    if (!coreThemeLink) {
+      const getExistingCoreThemeLinks = () => document.head.querySelectorAll('link[data-paragon-theme-core="true"]');
+      const removeExistingLinks = (existingLinks) => {
+        existingLinks.forEach((link) => {
+          link.remove();
+        });
+      };
+      // find existing links
+      const existingLinks = getExistingCoreThemeLinks();
+
+      // create new link
+      const createCoreThemeLink = (url) => {
+        coreThemeLink = document.createElement('link');
+        coreThemeLink.href = url;
+        coreThemeLink.rel = 'stylesheet';
+        coreThemeLink.dataset.paragonThemeCore = true;
+        coreThemeLink.onload = () => {
+          onLoad();
+          removeExistingLinks(existingLinks);
```

Comment:
What happens if the `onLoad` takes a while and we don't get to remove these existing links?  Does it adversely affect the layout at all?  (I'd assume not, but here I am just being paranoid)

Reply:
@davidjoy If `coreThemeLink.onload` is still loading, its styles haven't been applied in the browser yet. given this, I don't believe this is a concern? Any previous existing links would (and should) remain until the new link is fully resolved; otherwise, there would be a Flash of Unstyled Content.

For similar reasons, the `removeExistingLinks` is called after the `onLoad()` to ensure the new stylesheet is applied before removing the old ones